spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkDataWrite.java

```diff
@@ -649,6 +650,39 @@ public void testCommitUnknownException() throws IOException {
     Assert.assertEquals("Result rows should match", records, actual);
   }
 
+  @Test
+  public void testWriteWithCaseSensitiveOption() throws IOException {
+    File parent = temp.newFolder(format.toString());
+    File location = new File(parent, "test");
+
+    HadoopTables tables = new HadoopTables(CONF);
+    Table table =
+        tables.create(
+            SCHEMA,
+            PartitionSpec.unpartitioned(),
+            ImmutableMap.of(TableProperties.SPARK_WRITE_ACCEPT_ANY_SCHEMA, "true"),
+            location.toString());
+
+    List<SimpleRecord1> records = Lists.newArrayList(new SimpleRecord1(1, "a"));
+    Dataset<Row> dataFrame = spark.createDataFrame(records, SimpleRecord1.class);
```

Comment:
Instead of making a new class here could we just can do some dataframe games like

```java
spark.createDataFrame(records, SimpleRecord.class).withColumnRenamed("id", "Id").withColumnRenamed("data", "DaTa")
```
```